### PR TITLE
support multiple BUILD files in vendor

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: gazel/config.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ type Cfg struct {
 	// whether to add "pkg-srcs" and "all-srcs" filegroups
 	// note that this operates on the entire tree (not just SrcsDirs) but skips anything matching SkippedPaths
 	AddSourcesRules bool
+	// whether to have multiple build files in vendor/ or just one.
+	MultiBUILDVendor bool
```
Comment (spxtr) [inline gazel/config.go]: nit: `MultiBUILDVendor` is a super strange name. Maybe `VendorMultipleBuildFiles`? A bit longer but not so strange.

**Comment by spxtr** [comment]
Other than the nit this LGTM.

**Comment by sttts** [comment]
I did something similar today (without knowing that this PR went in) in https://github.com/mikedanese/gazel/commit/9961ffbf7fedf965ec1d8efc98e5062917a117e3. I wished we would have had this during the last weeks. 80% of rebase conflicts during the k8s.io/apiserver refactorings were due to our huge `vendor/BUILD` file.

Any plan to activate this in kube soon?

**Comment by mikedanese** [comment]
@sttts in the next couple days. This change was motivated by numerous rebases in vendor/BUILD so hopefully this helps.

**Pull request merged**